provisioning/image_customization.go

```diff
@@ -30,28 +30,35 @@ const (
 	ironicAgentImage = "IRONIC_AGENT_IMAGE"
 )
 
-func setIronicBaseUrl(name string, config *metal3iov1alpha1.ProvisioningSpec) corev1.EnvVar {
+func setIronicBaseUrl(name string, info *ProvisioningInfo) corev1.EnvVar {
+	config := info.ProvConfig.Spec
 	if config.ProvisioningNetwork != metal3iov1alpha1.ProvisioningNetworkDisabled && !config.VirtualMediaViaExternalNetwork {
 		return corev1.EnvVar{
 			Name:  name,
 			Value: "http://" + config.ProvisioningIP,
 		}
 	} else {
+		hostIP, err := getPodHostIP(info.Client.CoreV1(), info.Namespace)
+		if err == nil && hostIP != "" {
```

Comment:
Shouldn't we let this error bubble up so that we will retry, instead of just setting a blank URL?

Reply:
The retry will automatically happen during the next reconcile loop. We might get an error when the metal3 pod is not available temporarily. So, my understanding is that retrying immediately is not going to help.

Reply:
This is still to be resolved.
https://github.com/openshift/image-customization-controller/pull/28 is the change to improve the error reporting, so we can continue to handle `hostIP == ""` like this, but we need to return err in the case where `err != nil`. It might be easier to pass the ironic IP in to `createImageCustomizationContainer()` instead of calling out from inside it.

Reply:
Made updates. Hopefully this is what you are looking for.